docs/sources/mimir/operators-guide/architecture/query-sharding/index.md

```diff
@@ -139,6 +139,23 @@ parallel by the query-frontend, multiplying the previously set value of
 `-querier.max-query-parallelism` by
 `-query-frontend.query-sharding-total-shards`.
 
+## Cardinality estimation for query sharding (experimental)
+
+Since a high number of parallel sharded queries increases load on queriers and their dependencies, it is desirable to shard queries only as much as necessary.
+Queries that return more series, such as those that are of high cardinality, need to fetch more data and should therefore be split into a larger batch of shards.
+Queries that return few or no series should be executed with fewer or no shards at all.
+When determining the number of shards to use for a given query, the sharding logic can optionally take into account the cardinality observed during previous executions of the same query for similar time ranges.
```

Comment:
@osg-grafana Is it fine to clarify what cardinality is, this way?

```suggestion
When determining the number of shards to use for a given query, the sharding logic can optionally take into account the cardinality (number of series) observed during previous executions of the same query for similar time ranges.
```

Reply:
In the next generation of docs, we will have a topic about "What is cardinality?" that we can point people to. This current approach looks lean. Ship it. :)